packages/wallet/src/base-unlocked-wallet.ts

```diff
@@ -95,7 +95,7 @@ export class BaseWalletUnlocked extends BaseWalletLocked {
    */
   async simulateTransaction(transactionRequestLike: TransactionRequestLike): Promise<CallResult> {
     const transactionRequest = transactionRequestify(transactionRequestLike);
-
+    await this.provider.addMissingVariableOutputs(transactionRequest);
```

Comment:
the transactions in the wallets are signed.  I originally had it like this, call from the provider. But the UX was awkward because the provider had to re-sign the transaction; this felt like a big smell.